compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7025,6 +7037,30 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, HashSet<String> includedRecordFields,
+                                              List<String> includedRecordParamNames) {
+        Set<String> fields = ((BRecordType) Types.getReferredType(param.type)).fields.keySet();
+        fields.removeIf(field -> !includedRecordParamNames.contains(field));
```

Comment:
What's an example for when this will be true?

Reply:
Ex:
```ballerina
type Foo record {
    never a?;
    int b;
};

function callFoo(int a, *Foo foo) returns int {
    return a + foo.b;
}
```